packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -553,6 +555,113 @@ void main() {
 
     expect(newHeight, equals(initialHeight));
   });
+
+  testWidgets('Navigation indicator renders ripple', (WidgetTester tester) async {
+    final Widget widget = _buildWidget(
+      NavigationBar(
+        destinations: const <Widget>[
+          NavigationDestination(
+            icon: Icon(Icons.ac_unit),
+            label: 'AC',
+          ),
+          NavigationDestination(
+            icon: Icon(Icons.access_alarm),
+            label: 'Alarm',
+          ),
+        ],
+        onDestinationSelected: (int i) {
+        },
+      ),
+    );
+
+    await tester.pumpWidget(widget);
+
+    await tester.tap(find.byIcon(Icons.access_alarm));
+    await tester.pump(const Duration(milliseconds: 200));
+
+    final RenderObject inkFeatures = tester.allRenderObjects.firstWhere((RenderObject object) => object.runtimeType.toString() == '_RenderInkFeatures');
```

Comment:
Should we be checking that a rounded rectangle of the right size is being drawn?

Reply:
There is no rounded rectangle draw since `customBorder` simply adds clipping to `Box.circle` splash shape

So I've updated the test to check `paints..clipPath` with the indicator size. 
Wether it contains an indicator-sized clip-path or not.